Source/WebCore/loader/ContentFilter.cpp

```diff
@@ -123,6 +123,76 @@ bool ContentFilter::continueAfterWillSendRequest(ResourceRequest& request, const
     return !request.isNull();
 }
 
+ContentFilter::ContentFilterCallbackAggregator::~ContentFilterCallbackAggregator()
+{
+    ASSERT(RunLoop::isMain());
+    ASSERT(m_callback);
+
+    if (m_isBlocked) {
+        if (CheckedPtr contentFilter = m_contentFilter)
+            contentFilter->didDecide(State::Blocked);
+        m_callback(ResourceRequest());
+        return;
+    }
+
+    if (CheckedPtr contentFilter = m_contentFilter; m_numberOfFiltersAllowed == contentFilter->m_contentFilters.size())
+        contentFilter->didDecide(State::Allowed);
+
+    m_callback(WTFMove(m_request));
+}
+
+void ContentFilter::ContentFilterCallbackAggregator::didReceivePlatformContentFilterDecision(PlatformContentFilter& platformContentFilter, ResourceRequest&& request, PlatformContentFilter::State state)
```

Comment:
If we are not invoking `m_callback` until we receive decisions from all content filter, why do we need this function? Can't we just check the blocking state of each `PlatformContentFilter` in destructor `~ContentFilterCallbackAggregator`?